src/tls/openssl/tls_udp.c

```diff
@@ -242,18 +242,29 @@ static int tls_connect(struct tls_conn *tc)
 	r = SSL_connect(tc->ssl);
 	if (r <= 0) {
 		const int ssl_err = SSL_get_error(tc->ssl, r);
-
-		ERR_clear_error();
+		int err = 0;
 
 		switch (ssl_err) {
 
 		case SSL_ERROR_WANT_READ:
 			break;
+		
+		case SSL_ERROR_SYSCALL:
+		case SSL_ERROR_SSL:
+			DEBUG_WARNING("connect error: ");
+			ERR_print_errors_fp(stderr);
```

Comment:
one thing to consider, should the openssl errors be printed to STDERR or to the
DEBUG_xxx system? The Debug system has callback handlers so that the application
can redirect debug message to e.g. a file.